src/Microsoft.ApiDesignGuidelines.Analyzers/Core/EnumStorageShouldBeInt32.cs

```diff
@@ -25,25 +30,56 @@ public abstract class EnumStorageShouldBeInt32Analyzer : DiagnosticAnalyzer
-        internal static DiagnosticDescriptor NotIntegralRule = new DiagnosticDescriptor(RuleId,
-                                                                             s_localizableTitle,
-                                                                             s_localizableMessageNotIntegral,
-                                                                             DiagnosticCategory.Design,
-                                                                             DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
-                                                                             description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182147.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(NotInt32Rule, NotIntegralRule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(NotInt32Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            
+            analysisContext.RegisterCompilationStartAction(compilationContext =>
+            {
+                var flagsAttribute = WellKnownTypes.FlagsAttribute(compilationContext.Compilation);
+                compilationContext.RegisterSymbolAction(symbolContext => AnalyzeSymbol(symbolContext, flagsAttribute), SymbolKind.NamedType);
+            });
+        }
+
+        private void AnalyzeSymbol(SymbolAnalysisContext context, INamedTypeSymbol flagsAttribute)
+        {
+            var symbol = (INamedTypeSymbol)context.Symbol;
+
+            // If not enum type exit
+            if (symbol.TypeKind != TypeKind.Enum)
+            {
+                return;
+            }
+
+            // If enum is Int32 then exit
+            var underlyingType = symbol.EnumUnderlyingType.SpecialType;
+            if (underlyingType == SpecialType.System_Int32)
+            {
+                return;
+            }
+
+            // If accessibility of enum is not public exit
+            if (symbol.GetResultantVisibility() != SymbolVisibility.Public)
+            {
+                return;
+            }
+
+            // If enum is Int64 and has Flags attributes then exit
+            bool hasFlagsAttribute = symbol.GetAttributes().Any(a => a.AttributeClass.Equals(flagsAttribute));
+            if (underlyingType == SpecialType.System_Int64 && hasFlagsAttribute)
+            {
+                return;
+            }
+            else //Report diagnostic
```

Comment:
No need for the `else` or the braces; just fall through (as you have done above after all the other tests).